src/editor/hole.ts

```diff
@@ -41,6 +42,12 @@ export class Hole {
         if (code instanceof EditableTextTkn || code instanceof IdentifierTkn) {
             this.element.classList.add(Hole.editableHoleClass);
 
+            if(code instanceof IdentifierTkn && code.getParentStatement() instanceof ForStatement){
```

Comment:
Whenever we change the <code>IdentifierTkn</code> within a <code>ForStmt</code> we need to update its <code>loopVar</code> as well.